### PR TITLE
Fix `rubocop` in `brew audit`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

cc @ilovezfs

### CODE REVIEW

**Comment by ilovezfs** [comment]
Any idea why this wasn't caught by `brew tests` originally?

File: Library/Homebrew/cmd/style.rb [hunk 1]
```diff
@@ -120,7 +120,7 @@ def check_style_impl(files, output_type, options = {})
       # exitstatus can also be nil if RuboCop process crashes, e.g. due to
       # native extension problems.
       # JSON needs to be at least 2 characters.
-      if !status.success? || json.to_s.length < 2
+      if status.exitstatus > 1|| json.to_s.length < 2
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/style.rb]: Can you explain the difference here?

Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/style.rb]: @reitermarkus As-is it looks like a non-zero exit code is being considered a success which seems like a pretty big Ruby bug.

Comment (JCount) [inline Library/Homebrew/cmd/style.rb]: Rubocop exits with a 1 status code if offenses are found and a 0 exit code if there are no offenses. A convenient way of handling this in compact code might be to use:
```
if !status.exitstatus === 1..0 || json.to_s.length < 2
```

Comment (ilovezfs) [inline Library/Homebrew/cmd/style.rb]: It's worth noting what the old code was I think: https://github.com/Homebrew/brew/pull/2690/files

Why is `if $?.exitstatus.nil? || $?.exitstatus > 1 || json.to_s.length < 2` no loner appropriate?

Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/style.rb]: Ah, I see, thanks @reitermarkus. Agreed with @ilovezfs it may be worth adding the nil check.

File: Library/Homebrew/cmd/style.rb [hunk 2]
```diff
@@ -101,7 +101,7 @@ def check_style_impl(files, output_type, options = {})
       args << "--config" << HOMEBREW_LIBRARY_PATH/".rubocop.yml"
       args << HOMEBREW_LIBRARY_PATH
     else
-      args << "--config" << HOMEBREW_LIBRARY/".rubocop.yml"
+      args << "--config" << HOMEBREW_LIBRARY_PATH.parent/".rubocop.yml"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/style.rb]: Why this change?

File: Library/Homebrew/cmd/style.rb [hunk 3]
```diff
@@ -120,7 +120,7 @@ def check_style_impl(files, output_type, options = {})
       # exitstatus can also be nil if RuboCop process crashes, e.g. due to
       # native extension problems.
       # JSON needs to be at least 2 characters.
-      if !status.success? || json.to_s.length < 2
+      if !(0..1).cover?(status.exitstatus) || json.to_s.length < 2
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/style.rb]: Does this handle the `nil` case?

Comment (reitermarkus) [inline Library/Homebrew/cmd/style.rb]: Yep.

File: Library/.rubocop.yml [hunk 4]
```diff
@@ -1,4 +1,5 @@
 AllCops:
+  AllowSymlinksInCacheRootDirectory: true
```
Comment (MikeMcQuaid) [inline Library/.rubocop.yml]: What does this do?

File: Library/Homebrew/cmd/style.rb (hunk 2 above)
Comment (reitermarkus) [inline Library/Homebrew/cmd/style.rb]: `HOMEBREW_LIBRARY` is set to a temporary directory in tests.

File: Library/.rubocop.yml (hunk 4 above)
Comment (reitermarkus) [inline Library/.rubocop.yml]: Weird, now it is working even without it. I had to set it because RuboCop complained about `/tmp` being a symlink since the cache directory while running tests is inside `/tmp`.

File: Library/Homebrew/cmd/style.rb (hunk 2 above)
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/style.rb]: Feels a little weird to rely on this without at least a comment?

Comment (reitermarkus) [inline Library/Homebrew/cmd/style.rb]: We are already relying on this in the same way in the `.rubocop.yml` inside `HOMEBREW_LIBRARY_PATH`:

```yaml
inherit_from:
  - ../.rubocop.yml
```

Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/style.rb]: Sure, I just mean a comment explaining why we're doing this rather than `HOMEBREW_LIBRARY` would be good.

Comment (reitermarkus) [inline Library/Homebrew/cmd/style.rb]: Changed the test instead.

**Comment by ilovezfs** [comment]
@reitermarkus thanks for adding a test!

**Comment by JCount** [comment]
@reitermarkus Nice fix!
